src/main/java/fi/aalto/cs/intellij/services/PluginSettings.java

```diff
@@ -0,0 +1,55 @@
+package fi.aalto.cs.intellij.services;
+
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.serviceContainer.NonInjectable;
+
+import fi.aalto.cs.intellij.common.Course;
+import fi.aalto.cs.intellij.common.MalformedCourseConfigurationFileException;
+import fi.aalto.cs.intellij.notifications.CourseConfigurationError;
+import fi.aalto.cs.intellij.notifications.Notifier;
+
+import java.io.FileNotFoundException;
+
+import org.jetbrains.annotations.NotNull;
+
+public class PluginSettings {
+  @NotNull
+  private Course currentlyLoadedCourse;
```

Comment:
A possibility of a data race here? As a quick fix consider marking this `volatile`. We can think of more systematical synchronization approaches later. Or even more easily, just make it `final` until later.

Reply:
yes there is, i'll make it final for now, and then once we add the possibility of loading a different course, we'll consider synchronization